### PR TITLE
vowel challenge

### PR DESCRIPTION
**Pull request by SeR4h**
@dennisja please review the vowels based on the guidelines you gave me

### CODE REVIEW

File: vowel.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    set1.add(paragraph.charAt(i))
+    }
+    if(!characterCounter.hasOwnProperty(paragraph.charAt(i))){
+    characterCounter[paragraph.charAt(i)] = 1
+    //characterCounter[i]: myValue
+    //add it as key to my bject with value of 1
+    }
+    else if(characterCounter[(paragraph.charAt(i))] == 1){
+     counter =counter+ 1 
+    characterCounter[paragraph.charAt(i)] =characterCounter[paragraph.charAt(i)] +1
```
Comment (dennisja) [inline vowel.js]: Use the shorthand notation i.e
```js
// instead of 
x = x + 1

// do 
x += 1
```

File: vowel.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    for(var i=0;i<=paragraph.length;i++){
+    if(vowels.includes(paragraph.charAt(i))){
+    set1.add(paragraph.charAt(i))
+    }
+    if(!characterCounter.hasOwnProperty(paragraph.charAt(i))){
+    characterCounter[paragraph.charAt(i)] = 1
+    //characterCounter[i]: myValue
+    //add it as key to my bject with value of 1
+    }
+    else if(characterCounter[(paragraph.charAt(i))] == 1){
```
Comment (dennisja) [inline vowel.js]: Follow the indentation in the steps provided

File: vowel.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+
+function countVowels(paragraph){
```
Comment (dennisja) [inline vowel.js:2]: Cater for the spaces?

File: vowel.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    else if(characterCounter[(paragraph.charAt(i))] == 1){
+     counter =counter+ 1 
+    characterCounter[paragraph.charAt(i)] =characterCounter[paragraph.charAt(i)] +1
+    
+    }
+    }
+    
+    newVowel= Array.from(set1).join('')
+    var myArray= [newVowel,counter]
+    console.log(myArray)
```
Comment (dennisja) [inline vowel.js:29]: return myArray instead of logging it

File: vowel.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    //add it as key to my bject with value of 1
+    }
+    else if(characterCounter[(paragraph.charAt(i))] == 1){
+     counter =counter+ 1 
+    characterCounter[paragraph.charAt(i)] =characterCounter[paragraph.charAt(i)] +1
+    
+    }
+    }
+    
+    newVowel= Array.from(set1).join('')
```
Comment (dennisja) [inline vowel.js]: declare newVowel before use

**Review by dennisja — CHANGES_REQUESTED** [review]

**Comment by SeR4h** [comment]
@billkabanga could you please review my code. "#final vowel modification"

File: vowel.js [hunk 6]
```diff
@@ -3,28 +3,32 @@ function countVowels(paragraph){
...
     var myArray= [newVowel,counter]
     console.log(myArray)
     }
-    var s="This is a test string"
-    countVowels(s)
\ No newline at end of file
+    var str="This is a test string "
+    var finalstring = str.split(' ').join('')
+   
+    countVowels(finalstring)
```
Comment (billkabanga) [inline vowel.js]: Please add a new line at the end of the file

File: vowel.js [hunk 7]
```diff
@@ -3,28 +3,32 @@ function countVowels(paragraph){
...
+ 	if(characterCounter[(paragraph.charAt(i))] == 1){
+  
+ counter =counter+ 1
+     }
+    characterCounter[paragraph.charAt(i)] =characterCounter[paragraph.charAt(i)] +1  
     }
     }
-    
-    newVowel= Array.from(set1).join('')
+      console.log(characterCounter)
```
Comment (billkabanga) [inline vowel.js]: I suppose you need this `console.log` for debugging purposes, you should remove it when you are done using it.

File: vowel.js [hunk 8]
```diff
@@ -3,28 +3,32 @@ function countVowels(paragraph){
...
     //add it as key to my bject with value of 1
     }
-    else if(characterCounter[(paragraph.charAt(i))] == 1){
-     counter =counter+ 1 
-    characterCounter[paragraph.charAt(i)] =characterCounter[paragraph.charAt(i)] +1
-    
+    else{ 
+ 	if(characterCounter[(paragraph.charAt(i))] == 1){
+  
+ counter =counter+ 1
```
Comment (billkabanga) [inline vowel.js]: There should always be space around the operator

**Pull request merged**